src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientHandler.java

```diff
@@ -0,0 +1,121 @@
+   * @return If the client is valid
+   */
+  public static boolean isClientValid(SnowflakeStreamingIngestClient client) {
+    return client != null && !client.isClosed() && client.getName() != null;
+  }
+
+  private static final KCLogger LOGGER = new KCLogger(StreamingClientHandler.class.getName());
+  private static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT_";
+
+  private AtomicInteger createdClientId = new AtomicInteger(0);
+
+  /**
+   * Creates a streaming client from the given config
+   *
+   * @param connectorConfig The config to create the client
+   * @return A newly created client
+   */
+  public SnowflakeStreamingIngestClient createClient(Map<String, String> connectorConfig) {
+    LOGGER.info("Initializing Streaming Client...");
+
+    // get streaming properties from config
+    Properties streamingClientProps = new Properties();
+    streamingClientProps.putAll(
+        StreamingUtils.convertConfigForStreamingClient(new HashMap<>(connectorConfig)));
+
+    try {
+      // Override only if bdec version is explicitly set in config, default to the version set
+      // inside Ingest SDK
+      Map<String, Object> parameterOverrides = new HashMap<>();
+      Optional<String> snowpipeStreamingBdecVersion =
+          Optional.ofNullable(connectorConfig.get(SNOWPIPE_STREAMING_FILE_VERSION));
+      snowpipeStreamingBdecVersion.ifPresent(
+          overriddenValue -> {
+            LOGGER.info("Config is overridden for {} ", SNOWPIPE_STREAMING_FILE_VERSION);
+            parameterOverrides.put(BLOB_FORMAT_VERSION, overriddenValue);
+          });
+
+      String clientName = this.getNewClientName(connectorConfig);
+
+      SnowflakeStreamingIngestClient createdClient =
+          SnowflakeStreamingIngestClientFactory.builder(clientName)
+              .setProperties(streamingClientProps)
+              .setParameterOverrides(parameterOverrides)
+              .build();
+
+      LOGGER.info("Successfully initialized Streaming Client. ClientName:{}", clientName);
+
+      return createdClient;
+    } catch (SFException ex) {
+      LOGGER.error("Exception creating streamingIngestClient");
+      throw new ConnectException(ex);
+    }
+  }
+
+  /**
+   * Closes the given client. Swallows any exceptions
+   *
+   * @param client The client to be closed
+   */
+  public void closeClient(SnowflakeStreamingIngestClient client) {
+    // don't do anything if client is already invalid
+    if (!isClientValid(client)) {
+      return;
+    }
+
+    LOGGER.info("Closing Streaming Client:{}", client.getName());
+    try {
+      client.close();
+    } catch (Exception e) {
+      // the client should auto close, so don't throw an exception here
+      LOGGER.error(Utils.safeGetExceptionMessage("Failure closing Streaming client", e));
+    }
+  }
+
+  private String getNewClientName(Map<String, String> connectorConfig) {
+    return STREAMING_CLIENT_PREFIX_NAME
+        + connectorConfig.getOrDefault(Utils.NAME, "DEFAULT")
```

Comment:
nit: extract DEFAULT to a private static final variable. 
Also, this shouldnt happen since name is a required field, but just in case should we rename it to something more meaningful?

Reply:
I added the getOrDefault because tests were failing there. Thanks for pointing out its a required field - I've changed DEFAULT to TEST_CLIENT to better signal that it should only happen in test cases

Reply:
sounds good!